treeherder/webapp/api/push.py

```diff
@@ -71,6 +75,14 @@ def list(self, request, project):
 
             pushes = pushes.filter(repository=repository)
 
+        search_param = filter_params.get("search")
+        if search_param:       
+            pushes = pushes.annotate(
+                search=SearchVector(
```

Comment:
Is the search vector for the `push` needed? The same information is also part of the search vector for `commit`.

Reply:
This has been removed now.

Initially, I was facing an issue where indexing wasn't working on the query because the fields used on both `SearchVectors` were different. `Ginindex` searchvector fields for `Commit` were just `author` `comments` and `revision` but the query was prefixed with `commits__` because I wanted access to the `Commit` objects. To resolve it we thought adding an index on the `Push` model would work but it didn't.